spirv_hlsl.cpp

```diff
@@ -3147,6 +3164,18 @@ void CompilerHLSL::require_texture_query_variant(const SPIRType &type)
 	}
 }
 
+string CompilerHLSL::compile(std::vector<HLSLVertexAttr> *p_vertex_attributes)
```

Comment:
Better to take the vector by const<vector> &. There is no need for this to be a pointer.

Reply:
This is still a mutable pointer, this should be `const std::vector<HLSLVertexAttr> &` type.